### ISSUE TITLE
Only one header field with same name is returned

### ISSUE THREAD

**Issue by pilhuhn** [rest-client/rest-client#336]
According to some RFC, it is possible to have multiple header fields with the same name, that the client should treat like a single header field with that name, and with an array of values.

Suppose you have headers

```
$ curl -i -u rhqadmin:rhqadmin http://localhost:7080/rest/resource/platforms.json
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Pragma: No-cache
Cache-Control: no-cache
Expires: Thu, 01 Jan 1970 01:00:00 CET
Cache-Control: max-age=300, private
Link: <http://localhost:7080/rest/resource/platforms.json?page=-1>; rel="last"
Link: <http://localhost:7080/rest/resource/platforms.json>; rel="current"
X-collection-size: 2
Content-Type: application/json
Transfer-Encoding: chunked
Date: Fri, 28 Nov 2014 07:53:51 GMT
```

then `response.headers['link'] should return an array with 2 entries. But in fact when I have that above case

    print response.headers[:link].to_s + "\n"
    print response.raw_headers['link'].to_s + "\n"

yields

    <http://localhost:7080/rest/resource/platforms.json?page=-1>; rel="last"
    ["<http://localhost:7080/rest/resource/platforms.json?page=-1>; rel=\"last\"", "<http://localhost:7080/rest/resource/platforms.json>; rel=\"current\""]

So clearly there are  2 links that are  `raw_headers` but only 1 of them shows up in `headers`

**Comment by ab** [comment]
Fixed on 2.0.x branch.

Per RFC 7230, multiple headers in an HTTP response must be able to be combined by a comma on clients. (There is an exception for Set-Cookie, which violates this.)

You can still use `raw_headers` to get at the original arrays.

**Comment by pilhuhn** [comment]
Cool, thanks.
